.circleci/config.yml

```diff
@@ -1,8 +1,408 @@
+            export PATH="${PATH}:${GOPATH}/bin"
+            export ALGORAND_DEADLOCK=enable
+            GOLANG_VERSION=$(./scripts/get_golang_version.sh)
+            eval "$(~/gimme "${GOLANG_VERSION}")"
+            scripts/configure_dev.sh
+            scripts/buildtools/install_buildtools.sh -o "gotest.tools/gotestsum"
+            PACKAGES="$(go list ./... | grep -v /go-algorand/test/)"
+            export PACKAGE_NAMES=$(echo $PACKAGES | tr -d '\n')
+            export PARTITION_TOTAL=$CIRCLE_NODE_TOTAL
+            export PARTITION_ID=$CIRCLE_NODE_INDEX
+            export PARALLEL_FLAG="-p 1"
+            gotestsum --format testname --junitfile /tmp/results/<< parameters.result_subdir >>/results.xml --jsonfile /tmp/results/<< parameters.result_subdir >>/testresults.json -- --tags "sqlite_unlock_notify sqlite_omit_load_extension" -race -timeout 1h -coverprofile=coverage.txt -covermode=atomic -p 1 $PACKAGE_NAMES
+      - store_artifacts:
+          path: /tmp/results
+          destination: test-results
+      - store_test_results:
+          path: /tmp/results
+      - save_cache:
+          key: 'go-cache-{{ .Environment.CIRCLE_STAGE }}-{{ .Environment.CIRCLE_BUILD_NUM }}'
+          paths:
+            - tmp/go-cache
+
+  upload_coverage:
+    description: Collect coverage reports and upload them
+    steps:
+      - run:
+          name: Upload Coverage Reports
+          no_output_timeout: 10m
+          command: |
+            scripts/travis/upload_coverage.sh || true
+
+  generic_integration:
+    description: Run integration tests from build workspace, for re-use by diferent architectures
+    parameters:
+      circleci_home:
+        type: string
+        default: "/home/circleci"
+      result_subdir:
+        type: string
+    steps:
+      - attach_workspace:
+          at: << parameters.circleci_home >>
+      - run: mkdir -p /tmp/results/<< parameters.result_subdir >>
+      - run:
+          name: Run integration tests
+          no_output_timeout: 45m
+          command: |
+            set -x
+            export PATH=$(echo "$PATH" | sed -e 's|:<< parameters.circleci_home >>/\.go_workspace/bin||g' | sed -e 's|:/usr/local/go/bin||g')
+            export KMD_NOUSB=True
+            export GOPATH="<< parameters.circleci_home >>/go"
+            export PATH="${PATH}:${GOPATH}/bin"
+            export ALGORAND_DEADLOCK=enable
+            export BUILD_TYPE=integration
+            GOLANG_VERSION=$(./scripts/get_golang_version.sh)
+            eval "$(~/gimme "${GOLANG_VERSION}")"
+            scripts/configure_dev.sh
+            scripts/buildtools/install_buildtools.sh -o "gotest.tools/gotestsum"
+            export ALGOTEST=1
+            export SHORTTEST=
+            export TEST_RESULTS=/tmp/results/<< parameters.result_subdir >>
+            export PARTITION_TOTAL=$CIRCLE_NODE_TOTAL
+            export PARTITION_ID=$CIRCLE_NODE_INDEX
+            test/scripts/run_integration_tests.sh
```

Comment:
To avoid splitting the logic across multiple CI config files and make it easier to support our multiple build systems, all of these command blocks should be part of the existing CI scripts. We can rename the `scripts/travis` directory to `scripts/ci` and use conditions like:
```
if [ "$JENKINS_URL" != "" ];then

elif [ "$CI" = "true" ] && [ "$TRAVIS" = "true" ] && [ "$SHIPPABLE" != "true" ]; then

elif [ "$CI" = "true" ] && [ "$CIRCLECI" = "true" ]; then

fi
```